### PR TITLE
Reading markdown from file

### PR DESCRIPTION
**Pull request by jkylander**
Added some fancy markdown extenstions from https://facelessuser.github.io/pymdown-extensions/

Refactored html layout

### CODE REVIEW

File: challenges_project/settings.py
Comment (TypicallyThomas) [inline challenges_project/settings.py:1]: I'm not entirely sure what these deletions are supposed to accomplish. Can you explain?

File: challenges/challenges_app/urls.py
Comment (TypicallyThomas) [inline challenges/challenges_app/urls.py:1]: I'm assuming Django sanitises for XSS in URLs, but we'll need to check that to be sure we're not vulnerable by allowing users to use a string in the URL. I would personally prefer to use a Base 64 ID so we don't expose any weaknesses to XSS

File: challenges/challenges_app/util.py
Comment (TypicallyThomas) [inline challenges/challenges_app/util.py:1]: It's not a problem to just be lifting code from CS50 assignments but I'd suggest we don't copy on this large a scale. Runs the risk of making mistakes that are harder to spot, plus publishing the code on Github runs the risk of violating Academic Honesty

Comment (jkylander) [reply challenges/challenges_app/util.py:1]: it's part of the distribution code

**Review by TypicallyThomas — APPROVED** [review]

File: challenges/challenges_app/util.py
Comment (TypicallyThomas) [reply challenges/challenges_app/util.py:1]: Yeah, it's not a big problem in this case, just something to be wary of

File: challenges_project/settings.py
Comment (jkylander) [reply challenges_project/settings.py:1]: can't remember doing this :D

File: challenges/challenges_app/urls.py
Comment (jkylander) [reply challenges/challenges_app/urls.py:1]: at this point it is just trying to read a file from the directory, if there is no file, an error will show up. Not sure what kind of vuln that would be possible

Comment (TypicallyThomas) [reply challenges/challenges_app/urls.py:1]: Yeah, again just something we don't need to worry about too much right now but worth keeping an eye on

**Pull request merged**